service/service_test.go

```diff
@@ -0,0 +1,71 @@
+package service
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	cError "github.com/coreos/etcd/error"
+	"github.com/go-playground/assert/v2"
+	"gopkg.in/resty.v1"
+)
+
+func Test_PokemonAPI(t *testing.T) {
+	test := []struct {
+		name         string
+		statusCode   int
+		expectedBody []byte
+		id           string
+		err          error
+	}{
+		{
+			name:       "Pokemon found, response OK",
+			statusCode: http.StatusOK,
+			expectedBody: []byte(
+				`{
+				"id": 1,
+				"name": "bulbasaur",
+				}`,
+			),
+			id:  "1",
+			err: nil,
+		},
+		{
+			name:       "Pokemon not found, response Not found",
+			statusCode: http.StatusNotFound,
+			expectedBody: []byte(
+				`Not Found`,
+			),
+			id:  "100000",
+			err: cError.NewError(http.StatusNotFound, "error", 0),
+		},
+	}
+
+	for _, tt := range test {
+		client := resty.New().
+			SetHostURL("https://pokeapi.co/api/v2/").
```

Comment:
For testing API calls, you would like to use the following package to mock a sever https://golang.org/pkg/net/http/httptest/

Reply:
Done